test/server/http_filter_integration_test_base.cc

```diff
@@ -0,0 +1,63 @@
+#include "test/server/http_filter_integration_test_base.h"
+
+#include "server/well_known_headers.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+HttpFilterIntegrationTestBase::HttpFilterIntegrationTestBase(
+    Envoy::Network::Address::IpVersion ip_version)
+    : HttpIntegrationTest(Envoy::Http::CodecClient::Type::HTTP1, ip_version),
+      request_headers_({{":method", "GET"}, {":path", "/"}, {":authority", "host"}}) {}
+
+void HttpFilterIntegrationTestBase::initializeConfig(absl::string_view config) {
+  config_helper_.addFilter(std::string(config));
+  HttpIntegrationTest::initialize();
+}
+
+void HttpFilterIntegrationTestBase::updateRequestLevelConfiguration(
+    absl::string_view request_level_config) {
+  setRequestHeader(Server::TestServer::HeaderNames::get().TestServerConfig, request_level_config);
+}
+
+void HttpFilterIntegrationTestBase::switchToPostWithEntityBody() {
+  setRequestHeader(Envoy::Http::Headers::get().Method,
+                   Envoy::Http::Headers::get().MethodValues.Post);
+}
+
+void HttpFilterIntegrationTestBase::setRequestHeader(
+    const Envoy::Http::LowerCaseString& header_name, absl::string_view header_value) {
+  request_headers_.setCopy(header_name, header_value);
+}
+
+Envoy::IntegrationStreamDecoderPtr
+HttpFilterIntegrationTestBase::getResponse(ResponseOrigin expected_origin) {
+  cleanupUpstreamAndDownstream();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  Envoy::IntegrationStreamDecoderPtr response;
+  const bool is_post = request_headers_.Method()->value().getStringView() ==
+                       Envoy::Http::Headers::get().MethodValues.Post;
+  // Upon observing a POST request method, we inject a content body, as promised in the header file.
+  // This is useful, because emitting an entity body will hit distinct code in extensions. Hence we
+  // facilitate that.
+  const uint64_t request_body_size = is_post ? 1024 : 0;
+  // An extension can either act as an origin and synthesize a reply, or delegate that
```

Comment:
Looking through again, these comments are very helpful.  Thank you for adding them